packages/teams-js/src/public/video.ts

```diff
@@ -116,25 +207,124 @@ export namespace video {
    */
   export type VideoEffectCallback = (effectId: string | undefined) => Promise<void>;
 
+  type MediaStreamInfo = {
+    streamId: string;
+  };
+
   /**
-   * Register to read the video frames in Permissions section
+   * Register callbacks to process the video frames if the host supports it.
    * @beta
-   * @param frameCallback - The callback to invoke when registerForVideoFrame has completed
-   * @param config - VideoFrameConfig to customize generated video frame parameters
+   * @param parameters - Callbacks and configuration to process the video frames. A host may support either {@link MediaStreamCallback} or {@link SharedFrameCallback}, but not both.
+   * To ensure the video effect works on all supported hosts, the video app must provide both {@link MediaStreamCallback} and {@link SharedFrameCallback}.
+   * The host will choose the appropriate callback based on the host's capability.
+   *
+   * @example
+   * ```typescript
+   * video.registerForVideoFrame({
+   *   mediaStreamCallback: async (receivedVideoFrame) => {
+   *     const originalFrame = receivedVideoFrame.videoFrame as VideoFrame;
+   *     try {
+   *       const processedFrame = await processFrame(originalFrame);
+   *       return processedFrame;
+   *     } catch (e) {
+   *       throw e;
+   *     }
+   *   },
+   *   sharedFrameCallback: (
+   *     frame: VideoFrameData,
+   *     notifyVideoFrameProcessed: notifyVideoFrameProcessedFunctionType,
+   *     notifyError: notifyErrorFunctionType
+   *     ) => {
+   *       try {
+   *         processFrameInplace(frame);
+   *         notifyVideoFrameProcessed();
+   *       } catch (e) {
+   *         notifyError(e);
+   *       }
+   *     },
+   *   config: {
+   *     format: video.VideoPixelFormat.NV12,
+   *   }
+   * });
+   * ```
    */
-  export function registerForVideoFrame(frameCallback: VideoFrameCallback, config: VideoFrameConfig): void {
+  export function registerForVideoFrame(parameters: RegisterForVideoFrameParameters): void {
     ensureInitialized(runtime, FrameContexts.sidePanel);
     if (!isSupported()) {
       throw errorNotSupportedOnPlatform;
     }
+    if (!parameters.mediaStreamCallback || !parameters.sharedFrameCallback) {
+      throw new Error('Both mediaStreamCallback and sharedFrameCallback must be provided');
+    }
+    if (doesSupportMediaStream()) {
+      registerForMediaStream(parameters.mediaStreamCallback);
```

Comment:
Why `registerForMediaStream` ignores the `parameters.config` but `registerForSharedFrame` use the parameter?

Reply:
Currently config for `registerForMediaStream` is hard coded as `format: nv12`, we don't need any other config at the moment, let consider how to update this function when updating `videoEx` for avatar

Reply:
Discussed offline, there is no point to use 2 different configs for `registerForMediaStream` and `registerForSharedFrame`, updated to use the same config